### ISSUE TITLE
Mypy command failing when run through language server

### ISSUE THREAD

**Issue by ovangle** [microsoft/vscode-mypy#147]
When attempting to run mypy via vscode extension, I'm receiving an "unrecognised argument" error and no type errors are being reported.

The extension is logging that it is executing the following command:

```
<local-path-to-interpreter> -m mypy.dmypy --status-file /tmp/.vscode.dmypy_status/status-82977187-a80d-4822-b7b6-026891cdd011.json run -- --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-codes --no-pretty --config-file "./api/mypy.ini" <local-path-to-file>
````

`<local-path-to-interpreter>` is the correct interpreter for my python environment, and `<local-path-to-file>` is the correct absolute file path of the most recently saved file.

However, the mypy type checker extension logs the following usage error:

```
usage: mypy [-h] [-v] [-V] [more options; see below]
            [-m MODULE] [-p PACKAGE] [-c PROGRAM_TEXT] [files ...]
mypy: error: unrecognized arguments: <local-path-to-file>
```

When I copy and paste the command verbatim into the shell, mypy executes successfully and reports expected errors.

**Comment by karthiknadig** [comment]
@ovangle Would it be possible to share your settings?

**Comment by ovangle** [comment]
It's a pretty simple environment at the moment. I'm still bootstrapping this workspace.

`.vscode/settings.json` -- I don't know if it's possible to share the effective settings, but there is nothing python related overriden in either user or WSL remote settings. 
```
{
    "mypy-type-checker.importStrategy": "fromEnvironment",
    "mypy-type-checker.args": [
        "--config-file \"./api/mypy.ini\""
    ],
    "python.analysis.typeCheckingMode": "basic",
    "python.analysis.autoSearchPaths": false,
    "python.analysis.extraPaths": [
        "api/src/"
    ],
}
```

And `api/mypy.ini`
```
[mypy]
# warn_return_any = True
# warn_unused_configs = True
mypy_path = $MYPY_CONFIG_FILE_DIR/src
files = **/*.py
```

I'm using a python 3.11.2 virtualenv with the following packages
```
fastapi==0.101.1
pydantic-settings==2.0.3

SQLAlchemy[asyncio]==2.0.20
psycopg[c]===3.1.10
uvicorn[standard]==0.23.2

pyhumps==3.8.0

packaging==23.1
mypy==1.5.1
debugpy==1.7.0
```

Note that I had to add the `packaging` dependency after updating `mypy` extension to pre-release version (to fix `--show-error-code` problem) because it's called by `lsp_server.py` and wasn't in my virtualenv.

**Comment by karthiknadig** [comment]
The `mypy-type-checker.args` is setup wrong. Here is how you can fix it:
```
"mypy-type-checker.args": ["--config-file", "./api/mypy.ini"],
```
To ensure that `./api/mypy.ini` is always accessible. Add it like this:
```
"mypy-type-checker.args": ["--config-file", "${workspaceFolder}/api/mypy.ini"],
```

**Comment by ovangle** [comment]
Shell tokens :D Thanks a bunch.

I'd prefer to use mypy from environment, because that means I'm running the same using extension as I am from build tooling. However, `packaging` isn't a dependency of mypy, it's only used by `lsp_server.py` afaik. Dependencies which are used _only_ to execute the plugin should be bundled whether you're running from environment or otherwise? 

But otherwise, happy to close this, issue appears fixed :D

**Comment by ovangle** [comment]
Actually, I entered the

**Comment by karthiknadig** [comment]
> Dependencies which are used only to execute the plugin should be bundled whether you're running from environment or otherwise?

It is bundled. Can you share what you see early part of the logs where it says `sys.path`? there should be a reference to `bundled/libs` that includes `packaging`. It is added to the end of the `sys.path` as a fallback.

It should include packaging, as you see here:
![image](https://github.com/microsoft/vscode-mypy/assets/3840081/f022b08b-6fb2-49f8-bae7-ccbce0460982)


> So I'm just reopening to ask whether that would be something you can do something about, or whether vscode settings just turns everything which looks like an array into a list input?

This is handled by VS Code.

**Comment by karthiknadig** [comment]
The `packaging ` issue should be fixed by https://github.com/microsoft/vscode-mypy/pull/151

**Comment by ovangle** [comment]
Thanks for your help 👍
